src/photos/ducks/clustering/utils.js

```diff
@@ -12,22 +12,30 @@ export const prepareDataset = (photos, albums = []) => {
     .map(file => {
       const photo = {
         id: file._id || file.id,
-        name: file.name,
         clusterId: file.clusterId
       }
-      if (file.metadata) {
-        photo.datetime = file.metadata.datetime
-        photo.lat = file.metadata.gps ? file.metadata.gps.lat : null
-        photo.lon = file.metadata.gps ? file.metadata.gps.long : null
+      // Depending on the query, the attributes object might exists, or not
+      const attributes = file.attributes ? file.attributes : file
+      photo.name = attributes.name
+      const metadata = attributes.metadata
+      if (metadata) {
+        photo.datetime = metadata.datetime
+        photo.lat = metadata.gps ? metadata.gps.lat : null
+        photo.lon = metadata.gps ? metadata.gps.long : null
       } else {
-        photo.datetime = file.created_at
+        photo.datetime = attributes.created_at
       }
       const hours = new Date(photo.datetime).getTime() / 1000 / 3600
       photo.timestamp = hours
       // For each photo, we need to check the clusterid, i.e. the auto-album
       // referenced by the file. If there is none, the photo wasn't clustered before
-      if (!photo.clusterId && file.referenced_by) {
-        const ref = file.referenced_by.find(
+      if (
+        !photo.clusterId &&
+        file.relationships &&
+        file.relationships.referenced_by &&
+        file.relationships.referenced_by.data
```

Comment:
If we have lodash around, this sort of things can be simplified with `has(file, 'relationships.referenced_by.data')`

Reply:
It's actually more `get(file, 'relationships.referenced_by.data')` because `has` will pass if data is null, which we don't want